openwisp_firmware_upgrader/private_storage/views.py

```diff
@@ -12,9 +13,26 @@ class FirmwareImageDownloadView(PrivateStorageDetailView):
 
     def can_access_file(self, private_file):
         user = private_file.request.user
-        return user.is_superuser or (
+
+        # Check if user is superuser or manager of the organization
+        is_authorized = user.is_superuser or (
             user.is_staff and user.is_manager(self.object.build.category.organization)
         )
 
+        # If user is not authorized by role, deny access
+        if not is_authorized:
+            return False
+
+        # For authorized users, check view permission
+        # Only if they're not superusers (superusers have all permissions)
+        if not user.is_superuser:
+            perm_codename = get_permission_codename('view', self.model._meta)
+            view_perm = f'{self.model._meta.app_label}.{perm_codename}'
+            has_view_perm = user.has_perm(view_perm, self.object)
+            if not has_view_perm:
+                return False
```

Comment:
Did you consider using the [PermissionRequiredMixin](https://docs.djangoproject.com/en/5.2/topics/auth/default/#the-permissionrequiredmixin-mixin) instead of performing these checks manually?